options/m_option.c

```diff
@@ -2608,13 +2608,41 @@ static char *pretty_print_rel_time(const m_option_t *opt, const void *val)
     return print_rel_time(opt, val);
 }
 
+static int rel_time_set(const m_option_t *opt, void *dst,
+                        struct mpv_node *src)
+{
+    struct m_rel_time *t = dst;
+    if (src->format == MPV_FORMAT_STRING) {
+        return parse_rel_time(mp_null_log, opt, bstr0("rel_time_set"),
+                              bstr0(src->u.string), dst);
+    }
+    int rc = double_set(opt, &t->pos, src);
+    if (rc > 0)
+        t->type = REL_TIME_ABSOLUTE;
+    return rc;
+}
+
+static int rel_time_get(const m_option_t *opt, void *ta_parent,
+                        struct mpv_node *dst, void *src)
+{
+    struct m_rel_time *t = src;
+    if (t->type == REL_TIME_ABSOLUTE)
+        return double_get(opt, ta_parent, dst, &t->pos);
+
+    dst->format = MPV_FORMAT_STRING;
+    dst->u.string = print_rel_time(opt, t);
```

Comment:
Leaks memory. Allocations need to be chained to ta_parent for stupid reasons.

Reply:
Interesting, sorry for the beginner's mistake.
`ta_set_parent()` doesn't seem used much but looks like it does exactly what we need so I used it, if it's preferable not to use it I can make a common helper both print_rel_time and this will use that'll directly set the parent.